emsdk.py

```diff
@@ -1707,7 +1727,7 @@ def each_path_exists(pathlist):
       else:
         raise Exception('Unknown custom_is_installed_script directive "' + self.custom_is_installed_script + '"!')
 
-    return content_exists
+    return content_exists and (assume_is_installed_version or self.is_installed_version())
```

Comment:
I wonder we can just do `if not content_exist: return False` up on line 1715,

I guess it depends on what those `custom_is_installed_script` things do because they looks like they are currently independent of `content_exists`,

I looks me like if you move the content_exists calculation and the vs-tool handing below the custom_is_installed_script thing, then you can just return early when content_exists is false.

Reply:
The comment
```
#vs-tool is a special tool since all versions must be installed to the
# same dir, so dir name will not differentiate the version.
```

Makes me thing we can take this further perhaps?  Maybe a followup?

Reply:
Maybe possible in a follow. Like you, I'm not sure what those actually are.